torch_tvm/compiler.cpp

```diff
@@ -222,29 +233,40 @@ void TVMCompiler::run(Stack& stack) {
     value_to_ivalue[value_input] = inputs[i];
   }
 
-  CompleteArgumentSpec spec{false, ArrayRef<IValue>(inputs)};
-
-  if (cache_.find(spec) == cache_.end()) {
+  if (!cache_ || (cache_ && (*cache_).invalid)) {
     for (auto& kv : value_to_ivalue) {
-      kv.first->inferTypeFrom(kv.second.toTensor());
+      // TODO: convince Fuser to NOT create TVMCompilationGroups
+      // if ANY of subgraph inputs weren't profiled
+      TORCH_INTERNAL_ASSERT(kv.first->type()->cast<ProfiledTensorType>());
     }
     // bail out mechanism: try to convert to Relay, if it fails to convert the
     // graph by any reason(i.e. op difference), depend on the user preference,
     // either throw or fall back to the JIT interpreter for execution
+    cache_ = TVMObject {};
```

Comment:
at this point, if we run a graph twice with different sizes will we still get TVM compiled code each time?
is the logic for that moved up into the profiled executor?

Reply:
yup, we will bail out, profile again and generate a graph with a TVMCompGroups for different shapes.